viewing_party/party.py

```diff
@@ -1,23 +1,210 @@
+    '''
+
+    total_movies_watched = len(user_data["watched"])
+    rating_total = 0
+
+    for movie in user_data["watched"]:
+        rating_total += movie["rating"]
+
+    if total_movies_watched == 0:
+        average_rating = 0
+    else:
+        average_rating = rating_total/total_movies_watched
+
+    return average_rating
+
+def get_most_watched_genre(user_data):
+    '''
+    returns the genre most frequently listed in the watched list
+    '''
+    user_genre_count = {}
+
+    if len(user_data["watched"]) == 0:
+        most_watched_genre = None    
+    else:
+        for movie in user_data["watched"]:
+            genre = movie["genre"]
+            
+            if genre not in user_genre_count.keys():
+                user_genre_count[genre] = 1
+            else:
+                user_genre_count[genre] += 1
+        
+        most_watched_genre = max(user_genre_count, key=user_genre_count.get)
+    
+    return most_watched_genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+    '''
+    compares a user's watched movie list to their friends' watched movies lists 
+    and returns a list of unique movies the user has watched
+    '''
+
+    user_all_movies = []
+    friends_all_movies = []
+    user_unique_movies = []
+
+    for friend in user_data["friends"]:
+        for friend_movie in friend["watched"]:
+            friends_all_movies.append(friend_movie["title"])
+
+    for movie in user_data["watched"]:
+        user_all_movies.append(movie["title"])
+
+    for title in user_all_movies:
+        if not title in friends_all_movies:
+            unique_movie = title
+            for movie in user_data["watched"]:
+                if unique_movie == movie["title"]:
+                    user_unique_movies.append(movie)
+
+    return user_unique_movies
+
+def get_friends_unique_watched(user_data):
+    '''
+    compares a user's watched movie list to their friends' watched movies lists and
+    returns a list of unique movies the user's friends have watched
+    '''
+    user_all_movies = []
+    friends_all_movies = []
+    friends_unique_movies = []
+
+    for movie in user_data["watched"]:
+        user_all_movies.append(movie["title"])
+
+    for friend in user_data["friends"]:
+        for friend_movie in friend["watched"]:
+            if not friend_movie["title"] in friends_all_movies:
+                friends_all_movies.append(friend_movie["title"])
+
+    for title in friends_all_movies:
+        if not title in user_all_movies:
+            friend_unique_movie = title
+
+            for friend in user_data["friends"]:
+                for friend_movie in friend["watched"]:
+                    if friend_unique_movie == friend_movie["title"] \
+                        and not friend_movie in friends_unique_movies:
+                        friends_unique_movies.append(friend_movie)

```

Comment:
Similar to the previous function, this can be simplified by just using the dictionaries directly instead of extracting their titles.